pylint/config/option_manager_mixin.py

```diff
@@ -311,12 +292,44 @@ def read_config_file(self, config_file=None, verbose=None):
             msg = "No config file found, using default configuration"
         print(msg, file=sys.stderr)
 
+    @staticmethod
+    def _parse_toml(
+        config_file: Union[Path, str], parser: configparser.ConfigParser
+    ) -> None:
+        """Parse and handle errors of a toml configuration file."""
+        with open(config_file, encoding="utf-8") as fp:
+            content = toml.load(fp)
+        try:
+            sections_values = content["tool"]["pylint"]
+        except KeyError:
+            return
+        for section, values in sections_values.items():
+            # TOML has rich types, convert values to
+            # strings as ConfigParser expects.
+            if isinstance(values, list):
+                values = ",".join(values)
+            elif isinstance(values, dict):
+                for option, value in values.items():
+                    if isinstance(value, bool):
+                        values[option] = "yes" if value else "no"
+                    elif isinstance(value, (int, float)):
+                        values[option] = str(value)
+                    elif isinstance(value, list):
+                        values[option] = ",".join(value)
+                    else:
+                        values[option] = str(value)
+            parser._sections[section.upper()] = values  # type: ignore
+
     def load_config_file(self):
         """Dispatch values previously read from a configuration file to each
         options provider)"""
         parser = self.cfgfile_parser
         for section in parser.sections():
-            for option, value in parser.items(section):
+            try:
+                items = parser.items(section)
+            except ValueError:
+                self.add_message("bad-configuration-option-value", line=0, args=section)
```

Comment:
I think we need to return here? Otherwise we will handle half of the config if we find a bad value somewhere in the middle?

Reply:
Good catch, I think we need to continue to parse the rest of the configuration file.

Reply:
Do we though? It seems weird to continue parsing half of the file if we know we can't parse it fully? What if there are actual correct sections after the section that raises the `ValueError`? As a user I think I would expect a program to stop parsing an incorrect configuration instead of parsing half of it.

Reply:
But right now we're still running pylint after that, so if the program still runs I would expect the program to run with all the valid configurations it can parse. 

I agree that it would also make sense to raise a parsing error (``argparse.ArgumentParser.error(...)`` (we're not using argparse yet though) or cleanly raise and handle another error and print a nice message for it as it's not really a problem related to parsing anymore.

But we're not detecting a lot of parsing error for stuff option that do not exists but are in an existing section. We don't know what we expect in this configuration file. Isn't stopping the process for some errors and not the other would make pylint inconsistent ?  It's not like not taking things into account is particularly good either...

I'm half inclined to just fix the crash, release 2.12, and do that properly in 2.13.

Reply:
There is no easy way to keep continuing parsing the sections after the faulty section right? If that's not possible I think it is indeed perhaps best to fix this like this for 2.12 and improve handling in 2.13.

Reply:
I realized that for top level option we can sometime be certain that it's a bad option. I'm going to upgrade the MR to reflect that. This does not permit to detect bad value inside a valid top level section but is still better.